### PR TITLE
Add bucket to project relationship on satellite

### PR DESCRIPTION
**Pull request by rikysya**

### CODE REVIEW

File: cmd/storj-sim/network.go [hunk 1]
```diff
@@ -271,6 +292,217 @@ func newNetwork(flags *Flags) (*Processes, error) {
 	return processes, nil
 }
 
+type graphqlResponse struct {
```
Comment (egonelbre) [inline cmd/storj-sim/network.go]: let's put this logic in a separate file

File: cmd/storj-sim/network.go [hunk 2]
```diff
@@ -205,6 +211,21 @@ func newNetwork(flags *Flags) (*Processes, error) {
 			}
 
 			// TODO: maybe all the config flags should be exposed for all processes?
+			apiKey := vip.GetString("client.api-key")
+			if apiKey == "" {
+				parts := strings.Split(satellite.Address, ":")
```
Comment (egonelbre) [inline cmd/storj-sim/network.go]: Use https://golang.org/pkg/net/#SplitHostPort, using `strings.Split` doesn't work for IPv6 properly...

File: cmd/storj-sim/network.go [hunk 3]
```diff
@@ -205,6 +211,21 @@ func newNetwork(flags *Flags) (*Processes, error) {
...
+				consoleAddress := fmt.Sprintf("http://%s:%d/api/graphql/v0", parts[0], consolePort)
```
Comment (egonelbre) [inline cmd/storj-sim/network.go]: there can be multiple satellites and consoles running concurrently on the same machine

Comment (egonelbre) [reply cmd/storj-sim/network.go]: also users can later change both the satellite and console address in the config

File: cmd/storj-sim/console.go [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+	client := http.Client{}
+
+	var (
+		createUserFormat    = "mutation {createUser(input:{email:\"%s\",password:\"%s\",firstName:\"%s\",lastName:\"%s\"})}"
+		createProjectFormat = "mutation {createProject(input:{name:\"%s\",description:\"%s\"}){id}}"
+		createAPIKeyFormat  = "mutation {createAPIKey(projectID:\"%s\",name:\"%s\"){key}}"
+		tokenFormat         = "query {token(email:\"%s\",password:\"%s\"){token}}"
+	)
+
+	alice := console.CreateUser{
```
Comment (egonelbre) [inline cmd/storj-sim/console.go]: Not quite sure why these types are necessary here? I would probably hardcode the full queries everywhere. This is a temporary solution.

Comment (rikysya) [reply cmd/storj-sim/console.go]: Just to make code more clear, i guess

File: cmd/storj-sim/network.go [hunk 5]
```diff
@@ -206,6 +206,9 @@ func newNetwork(flags *Flags) (*Processes, error) {
 			}
 
 			// TODO: maybe all the config flags should be exposed for all processes?
+			// check if gateway config has an api key, if it's not
```
Comment (egonelbre) [inline cmd/storj-sim/network.go]: add space above so it's clear that it's a different message

File: satellite/console/consoleweb/consoleql/query_test.go [hunk 6]
```diff
@@ -197,21 +201,26 @@ func TestGraphqlQuery(t *testing.T) {
...
-		//_, err = service.ActivateAccount(ctx, activationToken1)
-		//if err != nil {
-		//	t.Fatal(err)
-		//}
-		//user1.Email = "muu1@email.com"
+		t.Run("Activation", func(t *testing.T) {
+			t.Skip("skip it until we will have activation flow ready")
+
+			//TODO(yar): skip it until we will have activation flow ready
+			//activationToken1, err := service.GenerateActivationToken(
```
Comment (egonelbre) [inline satellite/console/consoleweb/consoleql/query_test.go]: uncomment code

File: cmd/storj-sim/console.go [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+	defer func() {
+		err = resp.Body.Close()
+	}()
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return graphqlResponse{}, err
+	}
+
+	fmt.Println(string(b))
```
Comment (egonelbre) [inline cmd/storj-sim/console.go]: also remove the debug code

File: cmd/storj-sim/network.go [hunk 8]
```diff
@@ -205,6 +206,28 @@ func newNetwork(flags *Flags) (*Processes, error) {
...
+			// create example project with key and add it to the config
+			// so that gateway can have access to the satellite
+			apiKey := vip.GetString("client.api-key")
+			if apiKey == "" {
+				consoleAddress := fmt.Sprintf(
+					"http://%s/api/graphql/v0",
+					net.JoinHostPort(host, strconv.Itoa(consolePort+i)))
```
Comment (egonelbre) [inline cmd/storj-sim/network.go]: Lets's use vip.GetString to get the actual address of satellite.

**Pull request merged**